core/src/main/java/org/mskcc/cbio/portal/scripts/UpdateGenePanel.java

```diff
@@ -0,0 +1,142 @@
+package org.mskcc.cbio.portal.scripts;
+
+import org.mskcc.cbio.portal.model.*;
+import org.mskcc.cbio.portal.dao.*;
+import org.mskcc.cbio.portal.util.ProgressMonitor;
+
+import java.io.*;
+import java.util.*;
+import joptsimple.*;
+
+import static org.mskcc.cbio.portal.util.GenePanelUtil.extractGenes;
+import static org.mskcc.cbio.portal.util.GenePanelUtil.getPropertyValue;
+
+/**
+ * Updates an existing gene panel in a MySQL database.
+ * 
+ * @author jtquach1
+ */
+public class UpdateGenePanel extends ConsoleRunnable {
+
+    private File genePanelFile;
+
+    @Override
+    public void run() {
+        try {
+            String progName = "UpdateGenePanel";
+            String description = "Update gene panel files.";
+
+            OptionParser parser = new OptionParser();
+            OptionSpec<String> data = parser.accepts("data", "gene panel file").withRequiredArg()
+                    .describedAs("data_file.txt").ofType(String.class);
+            parser.accepts("noprogress",
+                    "this option can be given to avoid the messages regarding memory usage and % complete");
+
+            OptionSet options = null;
+            try {
+                options = parser.parse(args);
+            } catch (OptionException e) {
+                throw new UsageException(progName, description, parser, e.getMessage());
+            }
+            File genePanel_f = null;
+            if (options.has(data)) {
+                genePanel_f = new File(options.valueOf(data));
+            } else {
+                throw new UsageException(progName, description, parser, "'data' argument required.");
+            }
+
+            setFile(genePanel_f);
+            importData();
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void importData() throws Exception {
+        ProgressMonitor.setCurrentMessage("Reading data from:  " + genePanelFile.getAbsolutePath());
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(genePanelFile));
+
+        String stableId = getPropertyValue("stable_id", properties, true);
+        Set<CanonicalGene> canonicalGenes = extractGenes("gene_list", properties, true);
+        GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(stableId);
```

Comment:
Let's have some logic here that explicitly checks to make sure the gene panel is not null, and errors and exits if it is null

Reply:
Done with 
```java
        if (genePanel == null) {
            ProgressMonitor.logWarning("Gene panel " + stableId + " does not exist in the database! Exiting.");
            return;
        }
```